ide/gnatstudio/recordflux.py

```diff
@@ -0,0 +1,268 @@
+import re
+
+import GPS
+import highlighter.common as hl
+from gs_utils import hook
+
+XML = r"""<?xml version="1.0"?>
+<GNAT_Studio>
+   <Language>
+      <Name>RecordFlux</Name>
+      <Spec_Suffix>.rflx</Spec_Suffix>
+
+      <!-- Dummy body suffix as file templates don't work when its missing -->
+      <Body_Suffix>.workaround_rflx</Body_Suffix>
+
+      <Categories>
+         <Category>
+            <Name>package</Name>
+            <Pattern>^[ \\t]*package[ \\t]+((\\w|\\.)+)</Pattern>
+            <Index>1</Index>
+         </Category>
+         <Category>
+            <Name>type</Name>
+               <Pattern>^[ \\t]*type[ \\t]+(\\w+)</Pattern>
+               <Index>1</Index>
+         </Category>
+      </Categories>
+
+      <Context>
+         <New_Line_Comment_Start>--</New_Line_Comment_Start>
+         <String_Delimiter>&quot;</String_Delimiter>
+         <Can_Indent>True</Can_Indent>
+         <Syntax_Highlighting>True</Syntax_Highlighting>
+         <Case_Sensitive>False</Case_Sensitive>
+      </Context>
+   </Language>
+
+   <!-- Filter -->
+   <filter name="RecordFlux" language="RecordFlux"/>
+
+   <!-- Actions -->
+   <action name="check">
+      <filter_and>
+         <filter id="RecordFlux"/>
+         <filter id="Source editor"/>
+      </filter_and>
+      <external>rflx check %F</external>
+      <on-failure>
+         <shell lang="python" show-command="false">recordflux.parse_output(&quot;&quot;&quot;%1&quot;&quot;&quot;)</shell>
+      </on-failure>
+   </action>
+
+   <action name="check_all">
+      <filter id="File"/>
+      <shell lang="python" show-command="false">recordflux.get_source_files()</shell>
+      <external>rflx check %1"</external>
+      <on-failure>
+         <shell lang="python" show-command="false">recordflux.parse_output(&quot;&quot;&quot;%1&quot;&quot;&quot;)</shell>
+      </on-failure>
+   </action>
+
+   <action name="generate">
+      <filter_and>
+         <filter id="RecordFlux"/>
+         <filter id="Source editor"/>
+      </filter_and>
+      <external>rflx generate -d generated %F</external>
+      <on-failure>
+         <shell lang="python" show-command="false">recordflux.parse_output(&quot;&quot;&quot;%1&quot;&quot;&quot;)</shell>
+      </on-failure>
+   </action>
+
+   <action name="generate_all">
+      <filter_and>
+         <filter id="File"/>
+      </filter_and>
+      <shell lang="python" show-command="false">recordflux.get_source_files()</shell>
+      <external>rflx generate -d generated %1"</external>
+      <on-failure>
+         <shell lang="python" show-command="false">recordflux.parse_output(&quot;&quot;&quot;%1&quot;&quot;&quot;)</shell>
+      </on-failure>
+   </action>
+
+   <!-- Aliases -->
+   <alias name="rflx_package">
+      <param name="name" description="The name of the RecordFlux package"/>
+      <text>package %(name) is
+   %_
+end %(name);</text>
+   </alias>
+
+   <!-- Submenu -->
+   <submenu before="Window">
+      <title>RecordFlux</title>
```

Comment:
The ordering doesn't work for me: The RecordFlux menu is the last one (i.e. right of Help).

Reply:
Cf. T626-022

Reply:
Fixed in a09e199.